### DISCUSSION TITLE
Confusing error message with cascading option

### DISCUSSION THREAD

**Discussion by ViniciusAtaide** [prisma/prisma#15896] [Q&A] [answered]
I think I have an issue in my cascading dependencies, but the error message is cryptic and it points to an id that is already inside the main table which I'm trying to trucate with the deleteMany option. Have anyone been in this situation before?

```
Error:
Invalid `prisma.slot.deleteMany()` invocation:


Foreign key constraint failed on the field: `slot_id`
Serialized Error: {
  "clientVersion": "4.3.1",
  "code": "P2003",
  "meta": {
    "field_name": "slot_id",
  },
}
```

```
model Slot {
  id               Int            @id @default(autoincrement())
  attendanceType   String         @map("attendance_type") @db.VarChar(255)
  /// @zod.openapi({ type: 'string' })
  date             DateTime       @db.DateTime(0)
  duration         Int
  resourceId       Int            @map("resource_id")
  resourceConfigId Int            @map("resource_config_id")
  status           SlotStatus
  /// @zod.openapi({ type: 'string' })
  createdAt        DateTime       @default(now()) @map("created_at")
  /// @zod.openapi({ type: 'string' })
  updatedAt        DateTime       @updatedAt @map("updated_at")
  unitId           Int            @map("unit_id")
  resources        Resource       @relation(fields: [resourceId], references: [id], onDelete: Cascade, map: "slots_ibfk_1")
  resourceConfig   ResourceConfig @relation(fields: [resourceConfigId], references: [id], onDelete: Cascade, map: "slots_ibfk_2")
  schedules        Schedule[]

  @@index([resourceConfigId], map: "resourceConfigId")
  @@index([resourceId], map: "resourceId")
  @@index([date], map: "date")
  @@index([unitId], map: "fk_Slot_Unit1_idx")
}
```

**Comment by Rackar** [comment, marked as answer]
slot_id is necessary to related model, some records have this id. May be in you SlotStatus model.

**Reply by ViniciusAtaide** [reply]
I figured. The confusing part is that the error is hiding the related table. would love a feature to include it.

**Reply by nurul3101** [reply]
Would you mind opening a [Feature Request ](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)so that we could improve the error message?

**Reply by ViniciusAtaide** [reply]
absolutely my good sir

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
